pkg/deploy/controller/deployment/controller.go

```diff
@@ -116,16 +109,12 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		deploymentPod, err := c.podClient.createPod(deployment.Namespace, podTemplate)
 		// Retry on error.
 		if err != nil {
-			if config, decodeErr := c.decodeConfig(deployment); decodeErr == nil {
-				c.recorder.Eventf(config, kapi.EventTypeWarning, "FailedCreate", "Error creating deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
-			} else {
-				c.recorder.Eventf(deployment, kapi.EventTypeWarning, "FailedCreate", "Error creating deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
-			}
+			c.emitDeploymentEvent(deployment, config, kapi.EventTypeWarning, "FailedCreate", fmt.Sprintf("Error creating deployer pod", err))
```

Comment:
Nope

On Sun, May 1, 2016 at 2:21 AM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/deploy/controller/deployment/controller.go
> <https://github.com/openshift/origin/pull/8652#discussion_r61677340>:
>
> > @@ -116,16 +109,12 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
> >  		deploymentPod, err := c.podClient.createPod(deployment.Namespace, podTemplate)
> >  		// Retry on error.
> >  		if err != nil {
> > -			if config, decodeErr := c.decodeConfig(deployment); decodeErr == nil {
> > -				c.recorder.Eventf(config, kapi.EventTypeWarning, "FailedCreate", "Error creating deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
> > -			} else {
> > -				c.recorder.Eventf(deployment, kapi.EventTypeWarning, "FailedCreate", "Error creating deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
> > -			}
> > +			c.emitDeploymentEvent(deployment, config, kapi.EventTypeWarning, "FailedCreate", fmt.Sprintf("Error creating deployer pod", err))
>
> Any of the ones that are transient, probably not, *if* we generate an
> event when they would drop out of the retry queue. Do we do that today? On
> Apr 30, 2016, at 7:18 PM, Michail Kargakis <notifications@github.com>
> wrote: In pkg/deploy/controller/deployment/controller.go <#8652 (comment)
> <https://github.com/openshift/origin/pull/8652#discussion_r61676784>>:
> Do you find other failure modes useful for events here or should I remove
> emitting altogether? — You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <
> https://github.com/openshift/origin/pull/8652/files/7d9f3a30fbfe89487a3bc1322f79fb29f4ad4b68#r61676784
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8652/files/7d9f3a30fbfe89487a3bc1322f79fb29f4ad4b68#r61677340>
>